src/videojs-hls.js

```diff
@@ -995,7 +1019,7 @@ videojs.HlsHandler.prototype.loadSegment = function(segmentInfo) {
       return;
     }
 
-    self.lastSegmentLoaded_ = segmentInfo.uri;
+    self.lastSegmentLoaded_ = segment;
```

Comment:
@KatrinaE: actually, I do think checking the URI and byterange seems like the right move here (I'd really like to get rid of `lastSegmentLoaded_` entirely but I'm not sure how to yet). If you have bandwidth, we'd love a helping hand to get this merged. If you could switch this to the URI and byterange check instead of the object equality one, I think we're good to go.